internal/graph/check.go

```diff
@@ -485,7 +499,12 @@ func (c *LocalChecker) Close() {
 // was constructed with.
 func (c *LocalChecker) dispatch(_ context.Context, req *ResolveCheckRequest) CheckHandlerFunc {
 	return func(ctx context.Context) (*ResolveCheckResponse, error) {
-		return c.delegate.ResolveCheck(ctx, req)
+		resp, err := c.delegate.ResolveCheck(ctx, req)
+		if err != nil {
+			return resp, err
+		}
+		resp.GetResolutionMetadata().DispatchCount++
+		return resp, nil
```

Comment:
**Note:** I know this ruins the purity of the dispatch function but its a logical place to put this code.